In [2]:
import openai

In [3]:
openai.api_key = "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"

In [4]:
import pandas as pd
df = pd.read_csv("USEMedTrain.csv")
df
#Df contains all 1279747 enteries.

Unnamed: 0                                           Question  \
0                0  Which vitamin is supplied from only animal sou...   
1                1  All of the following are surgical options for ...   
2                2  Following endaerectomy on the right common car...   
3                3   Growth hormone has its effect on growth through?   
4                4     Scrub typhus is transmitted by: September 2004   
...            ...                                                ...   
182816      182816  Most common site for extra mammary Paget&;s di...   
182817      182817       Inferior Rib notching is seen in all except?   
182818      182818  Which is false regarding cryptococcus neoformans?   
182819      182819  Histopathological finding of gluten hypersensi...   
182820      182820  All except one are centrally acting muscle blo...   

                           Option1                              Option2  \
0                        Vitamin C                           Vitamin B7   
1       Adjustable gastric banding            Biliopancreatic diversion   
2       Central aery of the retina                    Infraorbital aery   
3                         Directly                                IG1-1   
4                            Louse                                 Tick   
...                            ...                                  ...   
182816                       Vulva                               Vagina   
182817          Coarctation of aoa  Classical blalock tausing operation   
182818   Grows at 5degC and 37degC                   It has 4 serotypes   
182819           Crypt hyperplasia  Increase in thickness of the mucosa   
182820                 Meprobamate                             Baclofen   

                             Option3                            Option4  \
0                        Vitamin B12                          Vitamin D   
1                    Duodenal Switch         Roux en Y Duodenal By pass   
2                      Lacrimal aery                 Nasociliary aretry   
3                          Thyroxine             Intranuclear receptors   
4                               Mite                               Milk   
...                              ...                                ...   
182816                         Penis                               Anus   
182817               SVC obstruction                  Neurofibromatosis   
182818               Urease negative  Causes superficial skin infection   
182819  Distal intestine involvement                Villous hypertrophy   
182820                      Diazepam                  Dantrolene sodium   

        Correct  
0             3  
1             4  
2             1  
3             2  
4             3  
...         ...  
182816        1  
182817        4  
182818        3  
182819        1  
182820        4  

[182821 rows x 7 columns]

In [5]:
df.size

1279747

# Table 1-  1000 questions

In [5]:
import pandas as pd

table1Question=[]
table1Correct=[]
table1Answer=[]

for i, row in df.iloc[:1000].iterrows():
    table1Question.append("Answer with a number only: " + row["Question"]+" 1."+ row["Option1"]+" 2."+ row["Option2"]+" 3."+ row["Option3"]+" 4."+ row["Option4"])
    table1Correct.append(row["Correct"])
    table1Answer.append("0")
#We have first 1000 enteries in table1

In [7]:
list_of_tuples = list(zip(table1Question,table1Correct,table1Answer))
table1 = pd.DataFrame(list_of_tuples, columns=["Question", "Correct", "Answer"])
table1

Question  Correct Answer
0    Answer with a number only: Which vitamin is su...        3      0
1    Answer with a number only: All of the followin...        4      0
2    Answer with a number only: Following endaerect...        1      0
3    Answer with a number only: Growth hormone has ...        2      0
4    Answer with a number only: Scrub typhus is tra...        3      0
..                                                 ...      ...    ...
995  Answer with a number only: Millenium developme...        2      0
996  Answer with a number only: In the stage of Gre...        4      0
997  Answer with a number only: Twin pregnancy, but...        2      0
998  Answer with a number only: Low bih weight baby...        4      0
999  Answer with a number only: Drug of choice for ...        2      0

[1000 rows x 3 columns]

# gpt-3.5-turbo

In [6]:
import openai  # for OpenAI API calls
from tenacity import (
    retry,
    stop_after_attempt,
    wait_random_exponential,
)  # for exponential backoff


@retry(wait=wait_random_exponential(min=1, max=60), stop=stop_after_attempt(6))
def completion_with_backoff(**kwargs):
    return openai.ChatCompletion.create(**kwargs)


In [9]:
for i in range(0, len(table1)):
    ans= completion_with_backoff(
    model = "gpt-3.5-turbo", 
    temperature = 0.8, 
    max_tokens = 2000,
    messages = [{"role": "system", "content": table1.loc[i,"Question"]}]
    )    
    table1.loc[i, ['Answer']] = ans.choices[0].message.content


In [10]:
table1

Question  \
0                                                                                                                                                                                                                                     Answer with a number only: Which vitamin is supplied from only animal source: 1.Vitamin C 2.Vitamin B7 3.Vitamin B12 4.Vitamin D   
1                                                                                                                                                             Answer with a number only: All of the following are surgical options for morbid obesity except - 1.Adjustable gastric banding 2.Biliopancreatic diversion 3.Duodenal Switch 4.Roux en Y Duodenal By pass   
2    Answer with a number only: Following endaerectomy on the right common carotid, a patient is found to be blind in the right eye. It is appears that a small thrombus embolized during surgery and lodged in the aery supplying the optic nerve. Which aery would be blocked? 1.Central aery of the retina 2.Infraorbital aery 3.Lacrimal aery 4.Nasociliary aretry   
3                                                                                                                                                                                                                                  Answer with a number only: Growth hormone has its effect on growth through? 1.Directly 2.IG1-1 3.Thyroxine 4.Intranuclear receptors   
4                                                                                                                                                                                                                                                               Answer with a number only: Scrub typhus is transmitted by: September 2004 1.Louse 2.Tick 3.Mite 4.Milk   
..                                                                                                                                                                                                                                                                                                                                                                 ...   
995                                                                                                                                                                                                                                     Answer with a number only: Millenium development goals formulated in 2000 were to be achieved by - 1.2005 2.2015 3.2010 4.2020   
996                                                                                                                                                                                              Answer with a number only: In the stage of Grey hepatization- 1.WBC's fill the alveoli 2.RBC's fill the alveoli 3.Organisms fill the alveoli 4.Accumulation of fibrin   
997                                                                                                                                                                                                           Answer with a number only: Twin pregnancy, but due to two different men is called: NEET 14 1.Superfetation 2.Superfecundation 3.Both of the above 4.None   
998                                                                                                                                                                                                                                 Answer with a number only: Low bih weight baby is defined as baby weighing: September 2006 1.1000 gm 2.1500 gm 3.2000 gm 4.2500 gm   
999                                                                                                                                                                                                                                                 Answer with a number only: Drug of choice for OCD is? 1.Clomipramine 2.Fluoxetine 3.Carbamezapine 4.Chlorpromazine   

     

In [11]:
for i in range(0, len(table1)):
    print(table1.loc[i, ['Answer']])
    

Answer    3
Name: 0, dtype: object
Answer    1
Name: 1, dtype: object
Answer    4. Nasociliary artery
Name: 2, dtype: object
Answer    2
Name: 3, dtype: object
Answer    3
Name: 4, dtype: object
Answer    4
Name: 5, dtype: object
Answer    3
Name: 6, dtype: object
Answer    2.bc
Name: 7, dtype: object
Answer    3. Deficit of expression by gesture
Name: 8, dtype: object
Answer    1
Name: 9, dtype: object
Answer    3. 2, 4, 5 True & 1, 3 false
Name: 10, dtype: object
Answer    3
Name: 11, dtype: object
Answer    4
Name: 12, dtype: object
Answer    3
Name: 13, dtype: object
Answer    3
Name: 14, dtype: object
Answer    1
Name: 15, dtype: object
Answer    3
Name: 16, dtype: object
Answer    2
Name: 17, dtype: object
Answer    2
Name: 18, dtype: object
Answer    3
Name: 19, dtype: object
Answer    2
Name: 20, dtype: object
Answer    1
Name: 21, dtype: object
Answer    3.0.9
Name: 22, dtype: object
Answer    3
Name: 23, dtype: object
Answer    2
Name: 24, dtype: object
Answer    2
Name: 25, 

Name: 299, dtype: object
Answer    4. None of the above
Name: 300, dtype: object
Answer    1
Name: 301, dtype: object
Answer    2
Name: 302, dtype: object
Answer    1
Name: 303, dtype: object
Answer    4
Name: 304, dtype: object
Answer    1. NADH-Q Oxidoreductase
Name: 305, dtype: object
Answer    4. Small airways are not involved
Name: 306, dtype: object
Answer    4. All
Name: 307, dtype: object
Answer    4
Name: 308, dtype: object
Answer    3
Name: 309, dtype: object
Answer    3. Ariculo temporal
Name: 310, dtype: object
Answer    1
Name: 311, dtype: object
Answer    4
Name: 312, dtype: object
Answer    4
Name: 313, dtype: object
Answer    4. Impaired liver function
Name: 314, dtype: object
Answer    3
Name: 315, dtype: object
Answer    4
Name: 316, dtype: object
Answer    1
Name: 317, dtype: object
Answer    4
Name: 318, dtype: object
Answer    4
Name: 319, dtype: object
Answer    4
Name: 320, dtype: object
Answer    1
Name: 321, dtype: object
Answer    4. Alkaline phosphatase
Name:

Name: 595, dtype: object
Answer    1. Almost total deafness
Name: 596, dtype: object
Answer    2
Name: 597, dtype: object
Answer    1. Albinism
Name: 598, dtype: object
Answer    1
Name: 599, dtype: object
Answer    4
Name: 600, dtype: object
Answer    4
Name: 601, dtype: object
Answer    2
Name: 602, dtype: object
Answer    1
Name: 603, dtype: object
Answer    4
Name: 604, dtype: object
Answer    2
Name: 605, dtype: object
Answer    4
Name: 606, dtype: object
Answer    2
Name: 607, dtype: object
Answer    2. Pelizaeus-Merzbacher disease
Name: 608, dtype: object
Answer    4
Name: 609, dtype: object
Answer    2
Name: 610, dtype: object
Answer    4
Name: 611, dtype: object
Answer    3
Name: 612, dtype: object
Answer    4
Name: 613, dtype: object
Answer    1
Name: 614, dtype: object
Answer    2. Atherosclerosis
Name: 615, dtype: object
Answer    2
Name: 616, dtype: object
Answer    1
Name: 617, dtype: object
Answer    2
Name: 618, dtype: object
Answer    3
Name: 619, dtype: object
Answer 

Name: 937, dtype: object
Answer    2
Name: 938, dtype: object
Answer    3
Name: 939, dtype: object
Answer    2
Name: 940, dtype: object
Answer    3
Name: 941, dtype: object
Answer    2
Name: 942, dtype: object
Answer    3
Name: 943, dtype: object
Answer    1
Name: 944, dtype: object
Answer    It is not possible to determine which aspect of the Thomas splint is most troubling without more information.
Name: 945, dtype: object
Answer    2.6%
Name: 946, dtype: object
Answer    5
Name: 947, dtype: object
Answer    2
Name: 948, dtype: object
Answer    4. Sphenoidal wing meningioma
Name: 949, dtype: object
Answer    3. 12-14 days
Name: 950, dtype: object
Answer    4. Spermatogonia
Name: 951, dtype: object
Answer    4
Name: 952, dtype: object
Answer    3
Name: 953, dtype: object
Answer    2
Name: 954, dtype: object
Answer    4
Name: 955, dtype: object
Answer    1
Name: 956, dtype: object
Answer    3.410 nm
Name: 957, dtype: object
Answer    4
Name: 958, dtype: object
Answer    4.
Name: 959, d

In [12]:
for i in range(0, len(table1)):
    table1.loc[i, ['Answer']] = table1.loc[i, ['Answer']][0][0]
table1    

Question  \
0                                                                                                                                                                                                                                     Answer with a number only: Which vitamin is supplied from only animal source: 1.Vitamin C 2.Vitamin B7 3.Vitamin B12 4.Vitamin D   
1                                                                                                                                                             Answer with a number only: All of the following are surgical options for morbid obesity except - 1.Adjustable gastric banding 2.Biliopancreatic diversion 3.Duodenal Switch 4.Roux en Y Duodenal By pass   
2    Answer with a number only: Following endaerectomy on the right common carotid, a patient is found to be blind in the right eye. It is appears that a small thrombus embolized during surgery and lodged in the aery supplying the optic nerve. Which aery would be blocked? 1.Central aery of the retina 2.Infraorbital aery 3.Lacrimal aery 4.Nasociliary aretry   
3                                                                                                                                                                                                                                  Answer with a number only: Growth hormone has its effect on growth through? 1.Directly 2.IG1-1 3.Thyroxine 4.Intranuclear receptors   
4                                                                                                                                                                                                                                                               Answer with a number only: Scrub typhus is transmitted by: September 2004 1.Louse 2.Tick 3.Mite 4.Milk   
..                                                                                                                                                                                                                                                                                                                                                                 ...   
995                                                                                                                                                                                                                                     Answer with a number only: Millenium development goals formulated in 2000 were to be achieved by - 1.2005 2.2015 3.2010 4.2020   
996                                                                                                                                                                                              Answer with a number only: In the stage of Grey hepatization- 1.WBC's fill the alveoli 2.RBC's fill the alveoli 3.Organisms fill the alveoli 4.Accumulation of fibrin   
997                                                                                                                                                                                                           Answer with a number only: Twin pregnancy, but due to two different men is called: NEET 14 1.Superfetation 2.Superfecundation 3.Both of the above 4.None   
998                                                                                                                                                                                                                                 Answer with a number only: Low bih weight baby is defined as baby weighing: September 2006 1.1000 gm 2.1500 gm 3.2000 gm 4.2500 gm   
999                                                                                                                                                                                                                                                 Answer with a number only: Drug of choice for OCD is? 1.Clomipramine 2.Fluoxetine 3.Carbamezapine 4.Chlorpromazine   

     

In [13]:
#table1 has 1000 questions solved using gpt-3.5-turbo
#Correct contains the right option and answer contains the option given by the model
table1.to_csv('Med1000GPT3.csv')

#  ada

In [36]:
import pandas as pd

table2Question=[]
table2Correct=[]
table2Answer=[]

for i, row in df.iloc[:1000].iterrows():
    table2Question.append("Answer with a number only: " + row["Question"]+" 1."+ row["Option1"]+" 2."+ row["Option2"]+" 3."+ row["Option3"]+" 4."+ row["Option4"])
    table2Correct.append(row["Correct"])
    table2Answer.append("0")
#We have first 1000 enteries in table2
list_of_tuples = list(zip(table2Question,table2Correct,table2Answer))
pd.set_option('display.max_colwidth', None)
table2 = pd.DataFrame(list_of_tuples, columns=["Question", "Correct", "Answer"])
table2


Question  \
0                                                                                                                                                                                                                                     Answer with a number only: Which vitamin is supplied from only animal source: 1.Vitamin C 2.Vitamin B7 3.Vitamin B12 4.Vitamin D   
1                                                                                                                                                             Answer with a number only: All of the following are surgical options for morbid obesity except - 1.Adjustable gastric banding 2.Biliopancreatic diversion 3.Duodenal Switch 4.Roux en Y Duodenal By pass   
2    Answer with a number only: Following endaerectomy on the right common carotid, a patient is found to be blind in the right eye. It is appears that a small thrombus embolized during surgery and lodged in the aery supplying the optic nerve. Which aery would be blocked? 1.Central aery of the retina 2.Infraorbital aery 3.Lacrimal aery 4.Nasociliary aretry   
3                                                                                                                                                                                                                                  Answer with a number only: Growth hormone has its effect on growth through? 1.Directly 2.IG1-1 3.Thyroxine 4.Intranuclear receptors   
4                                                                                                                                                                                                                                                               Answer with a number only: Scrub typhus is transmitted by: September 2004 1.Louse 2.Tick 3.Mite 4.Milk   
..                                                                                                                                                                                                                                                                                                                                                                 ...   
995                                                                                                                                                                                                                                     Answer with a number only: Millenium development goals formulated in 2000 were to be achieved by - 1.2005 2.2015 3.2010 4.2020   
996                                                                                                                                                                                              Answer with a number only: In the stage of Grey hepatization- 1.WBC's fill the alveoli 2.RBC's fill the alveoli 3.Organisms fill the alveoli 4.Accumulation of fibrin   
997                                                                                                                                                                                                           Answer with a number only: Twin pregnancy, but due to two different men is called: NEET 14 1.Superfetation 2.Superfecundation 3.Both of the above 4.None   
998                                                                                                                                                                                                                                 Answer with a number only: Low bih weight baby is defined as baby weighing: September 2006 1.1000 gm 2.1500 gm 3.2000 gm 4.2500 gm   
999                                                                                                                                                                                                                                                 Answer with a number only: Drug of choice for OCD is? 1.Clomipramine 2.Fluoxetine 3.Carbamezapine 4.Chlorpromazine   

     

In [38]:
for i in range(0, len(table2)):
    ans= openai.Completion.create(model = "text-ada-001", prompt=table3.loc[i,"Question"])
    table2.loc[i, ['Answer']] = ans['choices'][0]['text']
table2

Question  \
0                                                                                                                                                                                                                                     Answer with a number only: Which vitamin is supplied from only animal source: 1.Vitamin C 2.Vitamin B7 3.Vitamin B12 4.Vitamin D   
1                                                                                                                                                             Answer with a number only: All of the following are surgical options for morbid obesity except - 1.Adjustable gastric banding 2.Biliopancreatic diversion 3.Duodenal Switch 4.Roux en Y Duodenal By pass   
2    Answer with a number only: Following endaerectomy on the right common carotid, a patient is found to be blind in the right eye. It is appears that a small thrombus embolized during surgery and lodged in the aery supplying the optic nerve. Which aery would be blocked? 1.Central aery of the retina 2.Infraorbital aery 3.Lacrimal aery 4.Nasociliary aretry   
3                                                                                                                                                                                                                                  Answer with a number only: Growth hormone has its effect on growth through? 1.Directly 2.IG1-1 3.Thyroxine 4.Intranuclear receptors   
4                                                                                                                                                                                                                                                               Answer with a number only: Scrub typhus is transmitted by: September 2004 1.Louse 2.Tick 3.Mite 4.Milk   
..                                                                                                                                                                                                                                                                                                                                                                 ...   
995                                                                                                                                                                                                                                     Answer with a number only: Millenium development goals formulated in 2000 were to be achieved by - 1.2005 2.2015 3.2010 4.2020   
996                                                                                                                                                                                              Answer with a number only: In the stage of Grey hepatization- 1.WBC's fill the alveoli 2.RBC's fill the alveoli 3.Organisms fill the alveoli 4.Accumulation of fibrin   
997                                                                                                                                                                                                           Answer with a number only: Twin pregnancy, but due to two different men is called: NEET 14 1.Superfetation 2.Superfecundation 3.Both of the above 4.None   
998                                                                                                                                                                                                                                 Answer with a number only: Low bih weight baby is defined as baby weighing: September 2006 1.1000 gm 2.1500 gm 3.2000 gm 4.2500 gm   
999                                                                                                                                                                                                                                                 Answer with a number only: Drug of choice for OCD is? 1.Clomipramine 2.Fluoxetine 3.Carbamezapine 4.Chlorpromazine   

     

In [40]:
 for i in range(0, len(table2)):
    print(table2.loc[i, ['Answer']])

Answer    \n\n1.Vitamin C\n1.Vitamin B7\n1
Name: 0, dtype: object
Answer     means 5.Pulmonary hypertension\n All of the following are surgical options for
Name: 1, dtype: object
Answer     5.Ocular aery 6.vocal aery\n\n1.
Name: 2, dtype: object
Answer    \n\n1.Directly\n2.IG1-1\n3.
Name: 3, dtype: object
Answer    \n\n1. Louse 2. Tick 3. Mite 4. Milk
Name: 4, dtype: object
Answer    \n\n1.Punctation\n2.Mosaication\n3.
Name: 5, dtype: object
Answer    \n\n1. Neither is a good test for 1.Anal fiss
Name: 6, dtype: object
Answer     5.aac\n\nMetabolism:\n\nRemifenteuth
Name: 7, dtype: object
Answer    \n\n1.Hypomimia is ? 1.Decreased ability to
Name: 8, dtype: object
Answer    \n\n1.Clostridium tetani 2.Clostridium
Name: 9, dtype: object
Answer    \n\nFalse
Name: 10, dtype: object
Answer    \n\nThe pharmakokinetic change occurring in geriatric patient is due to
Name: 11, dtype: object
Answer    \n\n1. Time taken to adpt in the new environment is because of
Name: 12, dtype: object
Answer  

Name: 436, dtype: object
Answer    \n\n1.Heme\n2.Creatine phosphate\n3.Cy
Name: 437, dtype: object
Answer    \n\n Most common type of hypospadias is: \n\n1
Name: 438, dtype: object
Answer    \n\n1. Supscapular aery\n2. Thyroc
Name: 439, dtype: object
Answer    \n\n1.The lymphatics of the suprarenal gland detach into latter
Name: 440, dtype: object
Answer     5.Skull of Fibularnce 5.Conditioning the ball for use
Name: 441, dtype: object
Answer    \n\n\n1.The L5 nerve is located in the back of the
Name: 442, dtype: object
Answer    \n\n1.L-Alanine 2.L-Tyrosine 3
Name: 443, dtype: object
Answer    \n\nMirie disease is: 1.Migratory thromboph
Name: 444, dtype: object
Answer    . 5.T offline pacing.\n\n1. Continuous positive pressure breathing\n
Name: 445, dtype: object
Answer    \n\n1. Rectal polyp is most commonly presented as 1.Ob
Name: 446, dtype: object
Answer    \n\nUlipristal acetate is an oil-based product of the
Name: 447, dtype: object
Answer    \n\n1.HBV\n\n1.EBV\n\n1.
Name: 448, 

Name: 779, dtype: object
Answer     5.12 mm in width 6.1.2 mm in width 7.
Name: 780, dtype: object
Answer     1.2.3.\n\nVaccine
Name: 781, dtype: object
Answer    \n\nNo \n\ncarbamazepine
Name: 782, dtype: object
Answer    
Name: 783, dtype: object
Answer    \n\nBedside test for mental state-\n\n1.Four, main
Name: 784, dtype: object
Answer    ?\n\nThe suspect should beORGEously sick with a UTI and have
Name: 785, dtype: object
Answer    ’ gestation\n\n11
Name: 786, dtype: object
Answer     5.Lachryma sure 6.Lachrymal Missiles 7
Name: 787, dtype: object
Answer    ?\n\n1.Hemolytic jaundice\n2.C
Name: 788, dtype: object
Answer     5.It is a function of the study group\n\n1.It is
Name: 789, dtype: object
Answer    \n\n2
Name: 790, dtype: object
Answer    \n\nDobutamine has both b- and a-conotox
Name: 791, dtype: object
Answer    \n\n2. Sulphur 3. Phosphorus 4. Iod
Name: 792, dtype: object
Answer     5.Aorta disease\n\nA pair of Ehlers Danlos syndrome
Name: 793, dtype: object
Answer    \n\n

In [41]:
#cleaning the data:
#Find if the numbers 1, 2 , 3 or 4 are present in the answer?
for i in range(0, len(table2)):
    s=table2.loc[i, ['Answer']].astype("string")[0]
    if (s.find('1')!=-1):
        table2.loc[i, ['Answer']] = 1
    elif (s.find('2')!=-1):
        table2.loc[i, ['Answer']] = 2
    elif (s.find('3')!=-1):
        table2.loc[i, ['Answer']] = 3
    elif (s.find('4')!=-1):
        table2.loc[i, ['Answer']] = 4
    else: 
        table2.loc[i, ['Answer']] = 0
table2


Question  \
0                                                                                                                                                                                                                                     Answer with a number only: Which vitamin is supplied from only animal source: 1.Vitamin C 2.Vitamin B7 3.Vitamin B12 4.Vitamin D   
1                                                                                                                                                             Answer with a number only: All of the following are surgical options for morbid obesity except - 1.Adjustable gastric banding 2.Biliopancreatic diversion 3.Duodenal Switch 4.Roux en Y Duodenal By pass   
2    Answer with a number only: Following endaerectomy on the right common carotid, a patient is found to be blind in the right eye. It is appears that a small thrombus embolized during surgery and lodged in the aery supplying the optic nerve. Which aery would be blocked? 1.Central aery of the retina 2.Infraorbital aery 3.Lacrimal aery 4.Nasociliary aretry   
3                                                                                                                                                                                                                                  Answer with a number only: Growth hormone has its effect on growth through? 1.Directly 2.IG1-1 3.Thyroxine 4.Intranuclear receptors   
4                                                                                                                                                                                                                                                               Answer with a number only: Scrub typhus is transmitted by: September 2004 1.Louse 2.Tick 3.Mite 4.Milk   
..                                                                                                                                                                                                                                                                                                                                                                 ...   
995                                                                                                                                                                                                                                     Answer with a number only: Millenium development goals formulated in 2000 were to be achieved by - 1.2005 2.2015 3.2010 4.2020   
996                                                                                                                                                                                              Answer with a number only: In the stage of Grey hepatization- 1.WBC's fill the alveoli 2.RBC's fill the alveoli 3.Organisms fill the alveoli 4.Accumulation of fibrin   
997                                                                                                                                                                                                           Answer with a number only: Twin pregnancy, but due to two different men is called: NEET 14 1.Superfetation 2.Superfecundation 3.Both of the above 4.None   
998                                                                                                                                                                                                                                 Answer with a number only: Low bih weight baby is defined as baby weighing: September 2006 1.1000 gm 2.1500 gm 3.2000 gm 4.2500 gm   
999                                                                                                                                                                                                                                                 Answer with a number only: Drug of choice for OCD is? 1.Clomipramine 2.Fluoxetine 3.Carbamezapine 4.Chlorpromazine   

     

In [47]:
table2.to_csv('Med1000ada.csv')

Table 1 and 2: /v1/chat/completions endpoint
Models that can be used are : gpt-4, gpt-4-0613, gpt-4-32k, gpt-4-32k-0613, gpt-3.5-turbo, gpt-3.5-turbo-0613, gpt-3.5-turbo-16k, gpt-3.5-turbo-16k-0613

For the remaining tables of GPT 3 we use /v1/completions; 
models are text-davinci-003, text-davinci-002, text-curie-001, text-babbage-001, text-ada-001

# Table 3:  text-davinci-003 Very capable, faster ,2,049 tokens	


In [20]:
import pandas as pd

table3 = pd.DataFrame(list_of_tuples, columns=["Question", "Correct", "Answer"])
table3

Question  \
0                                                                                                                                                                                                                                     Answer with a number only: Which vitamin is supplied from only animal source: 1.Vitamin C 2.Vitamin B7 3.Vitamin B12 4.Vitamin D   
1                                                                                                                                                             Answer with a number only: All of the following are surgical options for morbid obesity except - 1.Adjustable gastric banding 2.Biliopancreatic diversion 3.Duodenal Switch 4.Roux en Y Duodenal By pass   
2    Answer with a number only: Following endaerectomy on the right common carotid, a patient is found to be blind in the right eye. It is appears that a small thrombus embolized during surgery and lodged in the aery supplying the optic nerve. Which aery would be blocked? 1.Central aery of the retina 2.Infraorbital aery 3.Lacrimal aery 4.Nasociliary aretry   
3                                                                                                                                                                                                                                  Answer with a number only: Growth hormone has its effect on growth through? 1.Directly 2.IG1-1 3.Thyroxine 4.Intranuclear receptors   
4                                                                                                                                                                                                                                                               Answer with a number only: Scrub typhus is transmitted by: September 2004 1.Louse 2.Tick 3.Mite 4.Milk   
..                                                                                                                                                                                                                                                                                                                                                                 ...   
995                                                                                                                                                                                                                                     Answer with a number only: Millenium development goals formulated in 2000 were to be achieved by - 1.2005 2.2015 3.2010 4.2020   
996                                                                                                                                                                                              Answer with a number only: In the stage of Grey hepatization- 1.WBC's fill the alveoli 2.RBC's fill the alveoli 3.Organisms fill the alveoli 4.Accumulation of fibrin   
997                                                                                                                                                                                                           Answer with a number only: Twin pregnancy, but due to two different men is called: NEET 14 1.Superfetation 2.Superfecundation 3.Both of the above 4.None   
998                                                                                                                                                                                                                                 Answer with a number only: Low bih weight baby is defined as baby weighing: September 2006 1.1000 gm 2.1500 gm 3.2000 gm 4.2500 gm   
999                                                                                                                                                                                                                                                 Answer with a number only: Drug of choice for OCD is? 1.Clomipramine 2.Fluoxetine 3.Carbamezapine 4.Chlorpromazine   

     

In [21]:
for i in range(0, len(table3)):
    ans= openai.Completion.create(model = "text-davinci-003", prompt=table3.loc[i,"Question"])
    table3.loc[i, ['Answer']] = ans['choices'][0]['text']
table3

Question  \
0                                                                                                                                                                                                                                     Answer with a number only: Which vitamin is supplied from only animal source: 1.Vitamin C 2.Vitamin B7 3.Vitamin B12 4.Vitamin D   
1                                                                                                                                                             Answer with a number only: All of the following are surgical options for morbid obesity except - 1.Adjustable gastric banding 2.Biliopancreatic diversion 3.Duodenal Switch 4.Roux en Y Duodenal By pass   
2    Answer with a number only: Following endaerectomy on the right common carotid, a patient is found to be blind in the right eye. It is appears that a small thrombus embolized during surgery and lodged in the aery supplying the optic nerve. Which aery would be blocked? 1.Central aery of the retina 2.Infraorbital aery 3.Lacrimal aery 4.Nasociliary aretry   
3                                                                                                                                                                                                                                  Answer with a number only: Growth hormone has its effect on growth through? 1.Directly 2.IG1-1 3.Thyroxine 4.Intranuclear receptors   
4                                                                                                                                                                                                                                                               Answer with a number only: Scrub typhus is transmitted by: September 2004 1.Louse 2.Tick 3.Mite 4.Milk   
..                                                                                                                                                                                                                                                                                                                                                                 ...   
995                                                                                                                                                                                                                                     Answer with a number only: Millenium development goals formulated in 2000 were to be achieved by - 1.2005 2.2015 3.2010 4.2020   
996                                                                                                                                                                                              Answer with a number only: In the stage of Grey hepatization- 1.WBC's fill the alveoli 2.RBC's fill the alveoli 3.Organisms fill the alveoli 4.Accumulation of fibrin   
997                                                                                                                                                                                                           Answer with a number only: Twin pregnancy, but due to two different men is called: NEET 14 1.Superfetation 2.Superfecundation 3.Both of the above 4.None   
998                                                                                                                                                                                                                                 Answer with a number only: Low bih weight baby is defined as baby weighing: September 2006 1.1000 gm 2.1500 gm 3.2000 gm 4.2500 gm   
999                                                                                                                                                                                                                                                 Answer with a number only: Drug of choice for OCD is? 1.Clomipramine 2.Fluoxetine 3.Carbamezapine 4.Chlorpromazine   

     

In [22]:
for i in range(0, len(table3)):
    print(table3.loc[i, ['Answer']])

Answer    \n\n3. Vitamin B12
Name: 0, dtype: object
Answer    \n\n4. Roux en Y Duodenal By pass
Name: 1, dtype: object
Answer    \n\n1.Central aery of the retina
Name: 2, dtype: object
Answer    \n\n1
Name: 3, dtype: object
Answer    \n\n2
Name: 4, dtype: object
Answer    \n\n1
Name: 5, dtype: object
Answer    \n\n1.No 2.No 3.No 4.No
Name: 6, dtype: object
Answer     5.abcd
Name: 7, dtype: object
Answer     5.Affective flattening\n\n5.Affective flattening
Name: 8, dtype: object
Answer    \n\n2
Name: 9, dtype: object
Answer    
Name: 10, dtype: object
Answer     \n\n3.Renal clearance
Name: 11, dtype: object
Answer    \n\n2
Name: 12, dtype: object
Answer    \n\n4. Urine microscopy for malignant cytology cells
Name: 13, dtype: object
Answer    \n\n3.Adenosine receptors
Name: 14, dtype: object
Answer    \n\n2
Name: 15, dtype: object
Answer    \n\n1
Name: 16, dtype: object
Answer    \n\n2.Setting
Name: 17, dtype: object
Answer    \n\n2
Name: 18, dtype: object
Answer    \n\n3
Name: 19, dtype

Name: 398, dtype: object
Answer    \n\n3. Allograft rejection
Name: 399, dtype: object
Answer    \n\n3.Late diastole
Name: 400, dtype: object
Answer    \n\n1
Name: 401, dtype: object
Answer     \n\n4
Name: 402, dtype: object
Answer    \n\n4
Name: 403, dtype: object
Answer    \n\n1. Panel discussion
Name: 404, dtype: object
Answer    \n\n3.Decreased insulin, increased free fatty acid which is not equivalent
Name: 405, dtype: object
Answer    \n\n4
Name: 406, dtype: object
Answer    \n\n4.Levodopa
Name: 407, dtype: object
Answer    \n\n3.Bambuterol
Name: 408, dtype: object
Answer     of the above\n\n4
Name: 409, dtype: object
Answer    
Name: 410, dtype: object
Answer    \n\n4. 50% Glycerine
Name: 411, dtype: object
Answer    \n\n2
Name: 412, dtype: object
Answer    \n\n2.Light chain globulins
Name: 413, dtype: object
Answer    \n\n1.Metformin
Name: 414, dtype: object
Answer    \n\n3.2007
Name: 415, dtype: object
Answer    \n\n2
Name: 416, dtype: object
Answer     due to pressure of forc

Name: 859, dtype: object
Answer    \n\n4. Kleihauer betke test
Name: 860, dtype: object
Answer    \n\n1. Birth Asphyxia
Name: 861, dtype: object
Answer    \n\n3. 60-100 L/min
Name: 862, dtype: object
Answer    \n\n1,2,4
Name: 863, dtype: object
Answer    \n\n3
Name: 864, dtype: object
Answer    \n\n1,2,3,4
Name: 865, dtype: object
Answer    \n\n4
Name: 866, dtype: object
Answer    \n\n2
Name: 867, dtype: object
Answer    \n\n3.Ecological study.
Name: 868, dtype: object
Answer    \n\n1.Endometriosis
Name: 869, dtype: object
Answer    \n\n4.Moderate persistent Asthma
Name: 870, dtype: object
Answer     5.e\n\n4.acd
Name: 871, dtype: object
Answer    \n\n4.Myelography
Name: 872, dtype: object
Answer    \n\n1.VIPoma
Name: 873, dtype: object
Answer    \n\n1, 2, 3, 4
Name: 874, dtype: object
Answer    \n\n3. Pressure
Name: 875, dtype: object
Answer    \n\n4
Name: 876, dtype: object
Answer    \n\n3.Delayed ejaculation
Name: 877, dtype: object
Answer    \n\n1. Zona pellucida
Name: 878, dtype: 

In [23]:
#cleaning the data:
#Find if the numbers 1, 2 , 3 or 4 are present in the answer?
for i in range(0, len(table3)):
    s=table3.loc[i, ['Answer']].astype("string")[0]
    if (s.find('1')!=-1):
        table3.loc[i, ['Answer']] = 1
    elif (s.find('2')!=-1):
        table3.loc[i, ['Answer']] = 2
    elif (s.find('3')!=-1):
        table3.loc[i, ['Answer']] = 3
    elif (s.find('4')!=-1):
        table3.loc[i, ['Answer']] = 4
    else: 
        table3.loc[i, ['Answer']] = 0
table3


Question  \
0                                                                                                                                                                                                                                     Answer with a number only: Which vitamin is supplied from only animal source: 1.Vitamin C 2.Vitamin B7 3.Vitamin B12 4.Vitamin D   
1                                                                                                                                                             Answer with a number only: All of the following are surgical options for morbid obesity except - 1.Adjustable gastric banding 2.Biliopancreatic diversion 3.Duodenal Switch 4.Roux en Y Duodenal By pass   
2    Answer with a number only: Following endaerectomy on the right common carotid, a patient is found to be blind in the right eye. It is appears that a small thrombus embolized during surgery and lodged in the aery supplying the optic nerve. Which aery would be blocked? 1.Central aery of the retina 2.Infraorbital aery 3.Lacrimal aery 4.Nasociliary aretry   
3                                                                                                                                                                                                                                  Answer with a number only: Growth hormone has its effect on growth through? 1.Directly 2.IG1-1 3.Thyroxine 4.Intranuclear receptors   
4                                                                                                                                                                                                                                                               Answer with a number only: Scrub typhus is transmitted by: September 2004 1.Louse 2.Tick 3.Mite 4.Milk   
..                                                                                                                                                                                                                                                                                                                                                                 ...   
995                                                                                                                                                                                                                                     Answer with a number only: Millenium development goals formulated in 2000 were to be achieved by - 1.2005 2.2015 3.2010 4.2020   
996                                                                                                                                                                                              Answer with a number only: In the stage of Grey hepatization- 1.WBC's fill the alveoli 2.RBC's fill the alveoli 3.Organisms fill the alveoli 4.Accumulation of fibrin   
997                                                                                                                                                                                                           Answer with a number only: Twin pregnancy, but due to two different men is called: NEET 14 1.Superfetation 2.Superfecundation 3.Both of the above 4.None   
998                                                                                                                                                                                                                                 Answer with a number only: Low bih weight baby is defined as baby weighing: September 2006 1.1000 gm 2.1500 gm 3.2000 gm 4.2500 gm   
999                                                                                                                                                                                                                                                 Answer with a number only: Drug of choice for OCD is? 1.Clomipramine 2.Fluoxetine 3.Carbamezapine 4.Chlorpromazine   

     

In [25]:
table3.to_csv('Med1000davinci.csv')

# Table 4: text-curie-001




In [26]:
table4 = pd.DataFrame(list_of_tuples, columns=["Question", "Correct", "Answer"])
table4

Question  \
0                                                                                                                                                                                                                                     Answer with a number only: Which vitamin is supplied from only animal source: 1.Vitamin C 2.Vitamin B7 3.Vitamin B12 4.Vitamin D   
1                                                                                                                                                             Answer with a number only: All of the following are surgical options for morbid obesity except - 1.Adjustable gastric banding 2.Biliopancreatic diversion 3.Duodenal Switch 4.Roux en Y Duodenal By pass   
2    Answer with a number only: Following endaerectomy on the right common carotid, a patient is found to be blind in the right eye. It is appears that a small thrombus embolized during surgery and lodged in the aery supplying the optic nerve. Which aery would be blocked? 1.Central aery of the retina 2.Infraorbital aery 3.Lacrimal aery 4.Nasociliary aretry   
3                                                                                                                                                                                                                                  Answer with a number only: Growth hormone has its effect on growth through? 1.Directly 2.IG1-1 3.Thyroxine 4.Intranuclear receptors   
4                                                                                                                                                                                                                                                               Answer with a number only: Scrub typhus is transmitted by: September 2004 1.Louse 2.Tick 3.Mite 4.Milk   
..                                                                                                                                                                                                                                                                                                                                                                 ...   
995                                                                                                                                                                                                                                     Answer with a number only: Millenium development goals formulated in 2000 were to be achieved by - 1.2005 2.2015 3.2010 4.2020   
996                                                                                                                                                                                              Answer with a number only: In the stage of Grey hepatization- 1.WBC's fill the alveoli 2.RBC's fill the alveoli 3.Organisms fill the alveoli 4.Accumulation of fibrin   
997                                                                                                                                                                                                           Answer with a number only: Twin pregnancy, but due to two different men is called: NEET 14 1.Superfetation 2.Superfecundation 3.Both of the above 4.None   
998                                                                                                                                                                                                                                 Answer with a number only: Low bih weight baby is defined as baby weighing: September 2006 1.1000 gm 2.1500 gm 3.2000 gm 4.2500 gm   
999                                                                                                                                                                                                                                                 Answer with a number only: Drug of choice for OCD is? 1.Clomipramine 2.Fluoxetine 3.Carbamezapine 4.Chlorpromazine   

     

In [27]:
for i in range(0, len(table4)):
    ans= openai.Completion.create(model = "text-curie-001", prompt=table4.loc[i,"Question"])
    table4.loc[i, ['Answer']] = ans['choices'][0]['text']
table4

Question  \
0                                                                                                                                                                                                                                     Answer with a number only: Which vitamin is supplied from only animal source: 1.Vitamin C 2.Vitamin B7 3.Vitamin B12 4.Vitamin D   
1                                                                                                                                                             Answer with a number only: All of the following are surgical options for morbid obesity except - 1.Adjustable gastric banding 2.Biliopancreatic diversion 3.Duodenal Switch 4.Roux en Y Duodenal By pass   
2    Answer with a number only: Following endaerectomy on the right common carotid, a patient is found to be blind in the right eye. It is appears that a small thrombus embolized during surgery and lodged in the aery supplying the optic nerve. Which aery would be blocked? 1.Central aery of the retina 2.Infraorbital aery 3.Lacrimal aery 4.Nasociliary aretry   
3                                                                                                                                                                                                                                  Answer with a number only: Growth hormone has its effect on growth through? 1.Directly 2.IG1-1 3.Thyroxine 4.Intranuclear receptors   
4                                                                                                                                                                                                                                                               Answer with a number only: Scrub typhus is transmitted by: September 2004 1.Louse 2.Tick 3.Mite 4.Milk   
..                                                                                                                                                                                                                                                                                                                                                                 ...   
995                                                                                                                                                                                                                                     Answer with a number only: Millenium development goals formulated in 2000 were to be achieved by - 1.2005 2.2015 3.2010 4.2020   
996                                                                                                                                                                                              Answer with a number only: In the stage of Grey hepatization- 1.WBC's fill the alveoli 2.RBC's fill the alveoli 3.Organisms fill the alveoli 4.Accumulation of fibrin   
997                                                                                                                                                                                                           Answer with a number only: Twin pregnancy, but due to two different men is called: NEET 14 1.Superfetation 2.Superfecundation 3.Both of the above 4.None   
998                                                                                                                                                                                                                                 Answer with a number only: Low bih weight baby is defined as baby weighing: September 2006 1.1000 gm 2.1500 gm 3.2000 gm 4.2500 gm   
999                                                                                                                                                                                                                                                 Answer with a number only: Drug of choice for OCD is? 1.Clomipramine 2.Fluoxetine 3.Carbamezapine 4.Chlorpromazine   

     

In [28]:
for i in range(0, len(table4)):
    s=table4.loc[i, ['Answer']].astype("string")[0]
    if (s.find('1')!=-1):
        table4.loc[i, ['Answer']] = 1
    elif (s.find('2')!=-1):
        table4.loc[i, ['Answer']] = 2
    elif (s.find('3')!=-1):
        table4.loc[i, ['Answer']] = 3
    elif (s.find('4')!=-1):
        table4.loc[i, ['Answer']] = 4
    else: 
        table4.loc[i, ['Answer']] = 0
table4

Question  \
0                                                                                                                                                                                                                                     Answer with a number only: Which vitamin is supplied from only animal source: 1.Vitamin C 2.Vitamin B7 3.Vitamin B12 4.Vitamin D   
1                                                                                                                                                             Answer with a number only: All of the following are surgical options for morbid obesity except - 1.Adjustable gastric banding 2.Biliopancreatic diversion 3.Duodenal Switch 4.Roux en Y Duodenal By pass   
2    Answer with a number only: Following endaerectomy on the right common carotid, a patient is found to be blind in the right eye. It is appears that a small thrombus embolized during surgery and lodged in the aery supplying the optic nerve. Which aery would be blocked? 1.Central aery of the retina 2.Infraorbital aery 3.Lacrimal aery 4.Nasociliary aretry   
3                                                                                                                                                                                                                                  Answer with a number only: Growth hormone has its effect on growth through? 1.Directly 2.IG1-1 3.Thyroxine 4.Intranuclear receptors   
4                                                                                                                                                                                                                                                               Answer with a number only: Scrub typhus is transmitted by: September 2004 1.Louse 2.Tick 3.Mite 4.Milk   
..                                                                                                                                                                                                                                                                                                                                                                 ...   
995                                                                                                                                                                                                                                     Answer with a number only: Millenium development goals formulated in 2000 were to be achieved by - 1.2005 2.2015 3.2010 4.2020   
996                                                                                                                                                                                              Answer with a number only: In the stage of Grey hepatization- 1.WBC's fill the alveoli 2.RBC's fill the alveoli 3.Organisms fill the alveoli 4.Accumulation of fibrin   
997                                                                                                                                                                                                           Answer with a number only: Twin pregnancy, but due to two different men is called: NEET 14 1.Superfetation 2.Superfecundation 3.Both of the above 4.None   
998                                                                                                                                                                                                                                 Answer with a number only: Low bih weight baby is defined as baby weighing: September 2006 1.1000 gm 2.1500 gm 3.2000 gm 4.2500 gm   
999                                                                                                                                                                                                                                                 Answer with a number only: Drug of choice for OCD is? 1.Clomipramine 2.Fluoxetine 3.Carbamezapine 4.Chlorpromazine   

     

In [29]:
table4.to_csv('Med1000Curie.csv') 

# Results

In [49]:
j=0
for i in range(0, len(table1)):
    ans=(table1.loc[i, ['Answer']][0])
    corr=(table1.loc[i, ['Correct']]).astype("string")[0]
    if(ans==corr):
        j=j+1
print(j)
print((j/1000)*100)
#3.5

542
54.2


In [50]:
j=0
for i in range(0, len(table2)):
    
    ans=(table2.loc[i, ['Answer']]).astype("string")[0]
    corr=(table2.loc[i, ['Correct']]).astype("string")[0]
    if(ans==corr):
        j=j+1
print(j)
print((j/1000)*100)
#ada

198
19.8


In [51]:
j=0
for i in range(0, len(table3)):
    ans=(table3.loc[i, ['Answer']]).astype("string")[0]
    corr=(table3.loc[i, ['Correct']]).astype("string")[0]
    if(ans==corr):
        j=j+1
print(j)
print((j/1000)*100)
#davinci

411
41.099999999999994


In [52]:
j=0
for i in range(0, len(table4)):
    ans=(table4.loc[i, ['Answer']]).astype("string")[0]
    corr=(table4.loc[i, ['Correct']]).astype("string")[0]
    if(ans==corr):
        j=j+1
print(j)
print((j/1000)*100)
#Curie

85
8.5


In [35]:
85/1000

0.085

# Results
Table 1 (gpt3.5-turbo): 286/500 =54.2%
Table 2 ( ) :
Table 3 (text-davinci-003): 185/500 =41.1%
Table 4 (text-curie-001): 59/500 =8.5%

In [8]:
table5 = pd.DataFrame(list_of_tuples, columns=["Question", "Correct", "Answer"])
table5

Question  Correct Answer
0    Answer with a number only: Which vitamin is su...        3      0
1    Answer with a number only: All of the followin...        4      0
2    Answer with a number only: Following endaerect...        1      0
3    Answer with a number only: Growth hormone has ...        2      0
4    Answer with a number only: Scrub typhus is tra...        3      0
..                                                 ...      ...    ...
995  Answer with a number only: Millenium developme...        2      0
996  Answer with a number only: In the stage of Gre...        4      0
997  Answer with a number only: Twin pregnancy, but...        2      0
998  Answer with a number only: Low bih weight baby...        4      0
999  Answer with a number only: Drug of choice for ...        2      0

[1000 rows x 3 columns]

In [12]:

#TRY GPT 4
import backoff  # for exponential backoff

@backoff.on_exception(backoff.expo, openai.error.RateLimitError)
def completions_with_backoff(**kwargs):
    return openai.Completion.create(**kwargs)

for i in range(0, len(table5)):
    ans= completion_with_backoff(
    model = "gpt-4", 
    temperature = 0.8, 
    max_tokens = 2000,
    messages = [{"role": "system", "content": table5.loc[i,"Question"]}]
    )    
    table5.loc[i, ['Answer']] = ans.choices[0].message.content
table5

RetryError: RetryError[<Future at 0x1e70a606b80 state=finished raised RateLimitError>]

In [13]:
table5[:60]

Question  Correct Answer
0   Answer with a number only: Which vitamin is su...        3      3
1   Answer with a number only: All of the followin...        4      4
2   Answer with a number only: Following endaerect...        1      1
3   Answer with a number only: Growth hormone has ...        2      2
4   Answer with a number only: Scrub typhus is tra...        3      3
5   Answer with a number only: Abnormal vascular p...        3      3
6   Answer with a number only: Per rectum examinat...        3      3
7   Answer with a number only: Characteristics of ...        3      4
8   Answer with a number only: Hypomimia is ? 1.De...        3      3
9   Answer with a number only: Naglers reaction is...        3      3
10  Answer with a number only: Which of the follow...        3      3
11  Answer with a number only: The pharmakokinetic...        3      3
12  Answer with a number only: True regarding lag ...        1      1
13  Answer with a number only: A 60 yr old chronic...        4      4
14  Answer with a number only: With which of the f...        3      3
15  Answer with a number only: Hyper viscosity is ...        1      4
16  Answer with a number only: For a positively sk...        1      1
17  Answer with a number only: The process of hard...        1      2
18  Answer with a number only: Superior vena cava ...        3      3
19  Answer with a number only: Testicular artery u...        1      1
20  Answer with a number only: Position of vocal c...        3      0
21  Answer with a number only: Organisms that has ...        2      0
22  Answer with a number only: Normal waist hip ra...        2      0
23  Answer with a number only: The most common cau...        3      0
24  Answer with a number only: A 6hours old snake ...        2      0
25  Answer with a number only: Which of the follow...        2      0
26  Answer with a number only: Treatment of choice...        2      0
27  Answer with a number only: What is the best me...        2      0
28  Answer with a number only: Lymph vessel which ...        1      0
29  Answer with a number only: Risk factors associ...        1      0
30  Answer with a number only: All are True about ...        3      0
31  Answer with a number only: Anterolateral ahros...        1      0
32  Answer with a number only: Regarding vaginal c...        3      0
33  Answer with a number only: 25 year old patient...        1      0
34  Answer with a number only: What is the most pr...        3      0
35  Answer with a number only: For TOF management ...        3      0
36  Answer with a number only: What change will be...        1      0
37  Answer with a number only: If solid line repre...        2      0
38  Answer with a number only: Pearsonian measure ...        2      0
39  Answer with a number only: Following are the c...        4      0
40  Answer with a number only: All of the followin...        4      0
41  Answer with a number only: During Sx for menin...        1      0
42  Answer with a number only: Which of the follow...        1      0
43  Answer with a number only: All of the followin...        4      0
44  Answer with a number only: Glucose sympo occur...        1      0
45  Answer with a number only: All are true about ...        3      0
46  Answer with a number only: Of the various moda...        1      0
47  Answer with a number only: In chronic alcoholi...        2      0
48  Answer with a number only: Minimum level of io...        3      0
49  Answer with a number only: True statement rega...        4      0
50  Answer with a number only: Pre-exposure prophy...        4      0
51  Answer with a number only: Regarding Sjogren's...        3      0
52  Answer with a number only: Best indicator of a...        1      0
53  Answer with a number only: Splenomegaly may be...        2      0
54  Answer with a number only: Gp2b3A inhibitors a...        4      0
55  Answer with a number only: True of case contro...        2      0
56  Answer with a number only: All of the followin...  

In [ ]:

for i in range(0, len(table2)):
    print(table5.loc[i, ['Answer']])


In [ ]:
#Saving only first digit of the answer
for i in range(0, len(table2)):
    table5.loc[i, ['Answer']] = table5.loc[i, ['Answer']][0][0]
table5

In [ ]:
table5.to_csv('table2.csv')   

In [7]:
#SAMPLE:
#TRY GPT 4
import backoff  # for exponential backoff

@backoff.on_exception(backoff.expo, openai.error.RateLimitError)
def completions_with_backoff(**kwargs):
    return openai.Completion.create(**kwargs)

ans= completion_with_backoff(
    model = "gpt-3.5-turbo", 
    temperature = 0.8, 
    max_tokens = 2000,
    messages = [{"role": "system", "content": "Answer with a letter only: For what period would archaeologists first begin to find permanent human settlements? A. The Paleolithic B. The Classical Era C. The Bronze Age D. The Neolithic"}]
    )   
ans

<OpenAIObject chat.completion id=chatcmpl-7owt6f7RQgpCjW3WGZojPfwEyxrzy at 0x26ea9973630> JSON: {
  "id": "chatcmpl-7owt6f7RQgpCjW3WGZojPfwEyxrzy",
  "object": "chat.completion",
  "created": 1692377120,
  "model": "gpt-3.5-turbo-0613",
  "choices": [
    {
      "index": 0,
      "message": {
        "role": "assistant",
        "content": "D. The Neolithic"
      },
      "finish_reason": "stop"
    }
  ],
  "usage": {
    "prompt_tokens": 47,
    "completion_tokens": 5,
    "total_tokens": 52
  }
}

In [9]:
type(ans)

openai.openai_object.OpenAIObject

In [10]:
ans

<OpenAIObject chat.completion id=chatcmpl-7owt6f7RQgpCjW3WGZojPfwEyxrzy at 0x26ea9973630> JSON: {
  "id": "chatcmpl-7owt6f7RQgpCjW3WGZojPfwEyxrzy",
  "object": "chat.completion",
  "created": 1692377120,
  "model": "gpt-3.5-turbo-0613",
  "choices": [
    {
      "index": 0,
      "message": {
        "role": "assistant",
        "content": "D. The Neolithic"
      },
      "finish_reason": "stop"
    }
  ],
  "usage": {
    "prompt_tokens": 47,
    "completion_tokens": 5,
    "total_tokens": 52
  }
}

In [11]:
ans.choices[0].message.content

'D. The Neolithic'